src/util/config.cpp

```diff
@@ -862,7 +862,8 @@ bool configt::set(const cmdlinet &cmdline)
       #endif
 
       // MinGW has extra defines
-      ansi_c.defines.push_back("__int64=\"long long\"");
+      // defines are shell quoted, which gives the wrong result in this case
+      ansi_c.preprocessor_options.push_back("-D__int64=\"long long\"");
     }
```

Comment:
This is a bug fix related to the PR, because it affects usability of the mingw build on windows. Due to shell quoting, using ansi_c.defines eventually produces code like this:
```
"long long" n;
```

Reply:
Can that be split out into a separate PR? There should be a proper solution for this that works for other defines as well.